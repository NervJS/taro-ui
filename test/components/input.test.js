import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtInput from '../../.temp/components/input/index'

describe('AtInput Snap', () => {
  it('render AtInput -- init', () => {
    const component = renderToString(<AtInput />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props style', () => {
    const component = renderToString(<AtInput style='border:none;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props value', () => {
    const component = renderToString(<AtInput value='value' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props name', () => {
    const component = renderToString(<AtInput name='name' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props name', () => {
    const component = renderToString(<AtInput placeholder='placeholder' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props title', () => {
    const component = renderToString(<AtInput title='title' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props maxlength', () => {
    const component = renderToString(<AtInput maxlength='10' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props type(number)', () => {
    const component = renderToString(<AtInput type='number' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props type(password)', () => {
    const component = renderToString(<AtInput type='password' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props type(idcard)', () => {
    const component = renderToString(<AtInput type='idcard' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props type(digit)', () => {
    const component = renderToString(<AtInput type='digit' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props type(phone)', () => {
    const component = renderToString(<AtInput type='phone' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props disabled', () => {
    const component = renderToString(<AtInput disabled />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props border', () => {
    const component = renderToString(<AtInput border={false} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props editable', () => {
    const component = renderToString(<AtInput editable={false} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props error', () => {
    const component = renderToString(<AtInput error />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props clear', () => {
    const component = renderToString(<AtInput clear />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInput -- props required', () => {
    const component = renderToString(<AtInput required />)
    expect(component).toMatchSnapshot()
  })
})
