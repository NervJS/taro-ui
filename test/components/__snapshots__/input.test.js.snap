// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtInput Snap render AtInput -- init 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props border 1`] = `"<div class=\\"at-input at-input--without-border\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props clear 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props disabled 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container at-input--disabled\\"><div class=\\"at-input__overlay\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props editable 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container at-input--disabled\\"><div class=\\"at-input__overlay\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props error 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container at-input--error\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefined<div class=\\"at-input__icon\\"><span class=\\"taro-text at-icon at-icon-alert-circle at-input__icon-alert\\"></span></div><div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props maxlength 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props name 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"name\\" name=\\"name\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props name 2`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"placeholder\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props required 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props style 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props title 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><label class=\\"at-input__title false\\" for=\\"\\">title</label><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props type(digit) 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"number\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props type(idcard) 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"idcard\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props type(number) 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"number\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props type(password) 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" password placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"password\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props type(phone) 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"11\\" value=\\"\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"11\\" type=\\"number\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;

exports[`AtInput Snap render AtInput -- props value 1`] = `"<div class=\\"at-input\\"><div class=\\"at-input__container\\"><div class=\\"at-input__overlay at-input__overlay--hidden\\"></div><input id=\\"\\" name=\\"\\" placeholderStyle=\\"\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"50\\" maxLength=\\"140\\" value=\\"value\\" confirmType=\\"done\\" cursor=\\"0\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" adjustPosition class=\\"weui-input at-input__input\\" placeholder=\\"\\" maxlength=\\"140\\" type=\\"text\\"/>undefinedundefined<div class=\\"at-input__children\\"></div></div></div>"`;
